### PR TITLE
Give the server config to the RoomWorkerStore

### PR DESCRIPTION
**Pull request by babolivier**

### CODE REVIEW

File: changelog.d/6509.bugfix [hunk 1]
```diff
@@ -0,0 +1 @@
+Fix pusher worker failing because it can't retrieve retention policies for rooms.
```
Comment (erikjohnston) [inline changelog.d/6509.bugfix]: Oh, ideally this would say "Fix regression where sending email push did not work when using a pusher worker" or somethibg

Comment (babolivier) [reply changelog.d/6509.bugfix]: Can't we fix that when generating the next changelog?

Comment (erikjohnston) [reply changelog.d/6509.bugfix]: It'd be nice to fix it up now, as otherwise it puts quite a large burden on those doing a release

Comment (babolivier) [reply changelog.d/6509.bugfix]: Done in https://github.com/matrix-org/synapse/commit/ae49d29ef1dabec25dc348bf11c2919d28d1cef3

**Pull request merged**